Description: Patch for Evisum
 This is a backport from efl 1.24 that prevents a seg fault in evisum v 0.5.9
 with efl 1.22.6
Author: Robert Wiley <ylee@bodhilinux.com>
Origin: other
Forwarded: no
Last-Update: 2021-01-10

--- a/src/lib/elementary/efl_ui_widget.c
+++ b/src/lib/elementary/efl_ui_widget.c
@@ -514,13 +514,13 @@ _full_eval(Eo *obj, Elm_Widget_Smart_Data *pd)
           {
              //emit signal and focus eval old and new
              ELM_WIDGET_DATA_GET(old_parent, old_pd);
-             _full_eval(old_parent, old_pd);
+             if (old_pd) _full_eval(old_parent, old_pd);
           }
 
         if (efl_isa(pd->logical.parent, EFL_UI_WIDGET_CLASS))
           {
              ELM_WIDGET_DATA_GET(pd->logical.parent, new_pd);
-             _full_eval(pd->logical.parent, new_pd);
+             if (new_pd) _full_eval(pd->logical.parent, new_pd);
           }
      }
 
@@ -552,7 +552,7 @@ _elm_widget_full_eval(Eo *obj)
 {
    ELM_WIDGET_DATA_GET(obj, pd);
 
-   _full_eval(obj, pd);
+   if (pd) _full_eval(obj, pd);
 }
 
 /**
@@ -641,6 +641,7 @@ _obj_mouse_down(void *data,
    Evas_Object *top;
 
    ELM_WIDGET_DATA_GET(data, sd);
+   if (!sd) return;
    Evas_Event_Mouse_Down *ev = event_info;
    if (ev->event_flags & EVAS_EVENT_FLAG_ON_HOLD) return;
 
@@ -656,6 +657,7 @@ _obj_mouse_move(void *data,
                 void *event_info)
 {
    ELM_WIDGET_DATA_GET(data, sd);
+   if (!sd) return;
    Evas_Event_Mouse_Move *ev = event_info;
    if (!sd->still_in) return;
 
@@ -677,6 +679,7 @@ _obj_mouse_up(void *data,
               void *event_info)
 {
    ELM_WIDGET_DATA_GET(data, sd);
+   if (!sd) return;
    Evas_Event_Mouse_Up *ev = event_info;
 
    if (sd->still_in && (ev->flags == EVAS_BUTTON_NONE) &&
@@ -693,6 +696,7 @@ _obj_mouse_in(void *data,
               void *event_info EINA_UNUSED)
 {
    ELM_WIDGET_DATA_GET(data, sd);
+   if (!sd) return;
    if (sd->focus_move_policy == ELM_FOCUS_MOVE_POLICY_IN)
      elm_widget_focus_mouse_up_handle(evas_object_widget_parent_find(obj));
 }
@@ -5438,6 +5442,7 @@ elm_widget_on_show_region_hook_set(Eo *obj, void *data, Efl_Ui_Scrollable_On_Sho
 {
    ELM_WIDGET_DATA_GET(obj, sd);
 
+   if (!sd) return;
    if ((sd->on_show_region_data == data) && (sd->on_show_region == func))
      return;
 
@@ -6110,6 +6115,7 @@ _efl_ui_model_property_bind_changed(void *data, const Efl_Event *event)
    const char *prop;
    unsigned int i;
 
+   if (!pd) return;
    EINA_ARRAY_ITER_NEXT(evt->changed_properties, i, prop, it)
      {
         Efl_Ui_Property_Bound *lookup;
@@ -6128,6 +6134,7 @@ _efl_ui_view_property_bind_changed(void *data, const Efl_Event *event)
    Eina_Stringshare *prop;
    unsigned int i;
 
+   if (!pd) return;
    EINA_ARRAY_ITER_NEXT(evt->changed_properties, i, prop, it)
      {
         Efl_Ui_Property_Bound *lookup;
